### DISCUSSION TITLE
test: Add end-to-end/integration tests with Playwright

### DISCUSSION THREAD

**Discussion by danorlando** [danny-avila/LibreChat#502] [Help Wanted]
Playwright has been set up, but E2E to tests still need to be written. This is an _excellent_ first issue has it allows an incoming developer to understand the application flow from end to end in order to make further contributions. If you are not familiar with Playwright, this is also a good opportunity to become familiar with an industry standard. If you are interested in taking this on but not sure where to start, talk to DanO on our discord server, as I would be more than happy to help you along. The following articles can be used for reference:

- [Build Automated Tests with Playwright](https://www.danorlandoblog.com/build-automated-test-suites-with-playwright/)
- [Debug End-to-End Tests Like a Pro with Playwright](https://www.danorlandoblog.com/debug-end-to-end-tests-like-a-pro-with-playwright/)

**Comment by danny-avila** [comment]
I wrote some instructions in the latest PR which should help setup the environment #344 

## To start writing frontend tests:

1. Run `npx install playwright` from the root directory
2. Start the App and make sure it is accessible via the default Url: `http://localhost:3080/`
3. Run `npm run e2e:auth` from the root directory
    - This will open a browser for you to save an authenticated state
    - Signup with a new account, for testing purposes (social logins will not work for testing)
    - Login, then Close the browser
    - You will now have the authenticated state saved in e2e/auth.json
4. Confirm tests run by running `npm run e2e`
5. Begin writing tests. Use the tests in `e2e/specs` as reference

**Comment by danny-avila** [comment]
If you would like to test messages, I wrote up a good first test for reference. Check it out in [/e2e/specs/messages.spec.js](https://github.com/danny-avila/chatgpt-clone/blob/fd5afc09a26aaf156735c7b1a3e8b8b287bceca6/e2e/specs/messages.spec.js)
